datasets/svhn.py

```diff
@@ -0,0 +1,489 @@
+    Format: 1 is used: Format with Bounding Boxes
+
+    Yuval Netzer, Tao Wang, Adam Coates, Alessandro Bissacco, Bo Wu, Andrew Y. Ng Reading Digits
+    in Natural Images with Unsupervised Feature Learning NIPS Workshop on Deep Learning and
+    Unsupervised Feature Learning 2011.
+    """
+
+    expansion_ratio = 0.3
+
+    def __init__(self, root_dir, d_type, transform=None, resize_size=(96, 96), fold_ratio=2,
+                 simplified=False):
+
+        if d_type not in ('test', 'train'):
+            raise ValueError("d_type can only be set to 'test' or 'train'")
+
+        if resize_size[0] != resize_size[1]:
+            raise ValueError('resize_size should be square')
+
+        self.root_dir = root_dir
+        self.d_type = d_type
+        self.transform = transform
+        self.resize_size = resize_size
+        self.fold_ratio = fold_ratio
+        self.simplified = simplified
+
+        self.img_list = []
+        self.boxes_list = []
+        self.lbls_list = []
+
+        self.processed_folder = os.path.join(root_dir, self.__class__.__name__, 'processed')
+        self.__makedir_exist_ok(self.processed_folder)
+
+        res_string = str(self.resize_size[0]) + 'x' + str(self.resize_size[1])
+        simplified_string = "_simplified" if self.simplified else ""
+
+        train_pkl_file_path = os.path.join(self.processed_folder, 'train_' + res_string +
+                                           '_fold_' + str(self.fold_ratio) + simplified_string +
+                                           '.pkl')
+        test_pkl_file_path = os.path.join(self.processed_folder, 'test_' + res_string + '_fold_' +
+                                          str(self.fold_ratio) + simplified_string + '.pkl')
+
+        if self.d_type == 'train':
+            self.pkl_file = train_pkl_file_path
+            self.info_df_csv_file = os.path.join(self.processed_folder, 'train_info.csv')
+        elif self.d_type == 'test':
+            self.pkl_file = test_pkl_file_path
+            self.info_df_csv_file = os.path.join(self.processed_folder, 'test_info.csv')
+        else:
+            print(f'Unknown data type: {self.d_type}')
+            return
+
+        self.__create_info_df_csv()
+        self.__create_pkl_file()
+        self.is_truncated = False
+
+    def __create_info_df_csv(self):
+
+        if os.path.exists(self.info_df_csv_file):
+            self.info_df = pd.read_csv(self.info_df_csv_file)
+
+            for column in self.info_df.columns:
+                if column in ['label', 'x0', 'x1', 'y0', 'y1']:
+                    self.info_df[column] = \
+                        self.info_df[column].apply(ast.literal_eval)
+        else:
+
+            mat_file_path = os.path.join(self.root_dir, self.__class__.__name__, self.d_type,
+                                         'digitStruct.mat')
+
+            if not os.path.exists(mat_file_path):
+                print('\nDownload the archieve file from: '
```

Comment:
Add "Review the terms and conditions on http://ufldl.stanford.edu/housenumbers/ and then download..."

Spelling: **archive**